commands/bake.go

```diff
@@ -588,3 +604,62 @@ func printTargetList(w io.Writer, cfg *bake.Config) error {
 
 	return nil
 }
+
+func bakeMetricAttributes(dockerCli command.Cli, driverType string, targets []string, options *bakeOptions) attribute.Set {
+	return attribute.NewSet(
+		commandNameAttribute.String("bake"),
+		attribute.Stringer(string(commandOptionsHash), &bakeOptionsHash{
+			bakeOptions: options,
+			configDir:   confutil.ConfigDir(dockerCli),
+			targets:     targets,
+		}),
+		driverNameAttribute.String(options.builder),
+		driverTypeAttribute.String(driverType),
+	)
+}
+
+type bakeOptionsHash struct {
+	*bakeOptions
+	configDir  string
+	targets    []string
+	result     string
+	resultOnce sync.Once
+}
+
+func (o *bakeOptionsHash) String() string {
+	o.resultOnce.Do(func() {
+		// Will be empty if no argument was specified
+		var files []string
+		if len(o.files) == 0 {
+			files = []string{osutil.GetWd()}
+		} else {
+			files = make([]string, len(o.files))
+			for i, file := range o.files {
+				files[i] = osutil.ToAbs(file)
+			}
+			sort.Strings(files)
+		}
```

Comment:
Not sure that would work for remote bake definitions, maybe using `cmdContext` would work.

Reply:
Can you expand on what you mean by this? I'm not very familiar with remote bake definitions. I'm guessing this is referring to `osutil.ToAbs`?

Reply:
> I'm guessing this is referring to `osutil.ToAbs`?

Yes I mean `osutil.*` calls that would not be right for remote bake definition set here: https://github.com/docker/buildx/blob/e2f6808457ab9d8de4a85f0f26b0028411ef20fc/commands/bake.go#L68 and retrieved here https://github.com/docker/buildx/blob/e2f6808457ab9d8de4a85f0f26b0028411ef20fc/commands/bake.go#L474

Stat of these definitions are not on the host but memory: https://github.com/docker/buildx/blob/e2f6808457ab9d8de4a85f0f26b0028411ef20fc/bake/remote.go#L186-L194

Hence why I think we should use `cmdContext` and just the relative path of `files`.

Reply:
Ok I think I understand now. Do you think that it would be better to just include the `url` and `cmdContext` attributes in the hash instead of trying to resolve the path of the files? The intention is just to differentiate different bake commands in different directories on the same machine. I don't think we need to resolve the absolute path to these files.

I'll mock something up and see how it looks.